extensions/amp-wpm-player/0.1/amp-wpm-player.js

```diff
@@ -0,0 +1,557 @@
+     * specified in the name parameter
+     * @private
+     * @param {string} name
+     * @param {boolean} required Specifies weather to throw and error when the attribute is not preset
+     *
+     * @return {string}
+     */
+    parseAttribute.string = (name, required) =>
+      parseAttribute(
+          name,
+          required,
+          value => value,
+      );
+
+    /**
+     * Method that parses a number from the html attribute
+     * specified in the name parameter
+     * @private
+     * @param {string} name
+     * @param {boolean} required Specifies weather to throw and error when the attribute is not preset
+     *
+     * @return {number}
+     */
+    parseAttribute.number = (name, required) =>
+      parseAttribute(
+          name,
+          required,
+          value => parseInt(value, 10),
+      );
+
+    const clip = parseAttribute.json('clip') || {};
+    return {
+      ampcontrols: true,
+      forceUrl4stat: this.win.location.href,
+      target: 'playerTarget',
+      ...clip,
+      adv: parseAttribute.boolean('adv'),
+      url: parseAttribute.string('url'),
+      title: parseAttribute.string('title'),
+      screenshot: parseAttribute.string('screenshot'),
+      forcerelated: parseAttribute.boolean('forcerelated'),
+      hiderelated: parseAttribute.boolean('hiderelated'),
+      hideendscreen: parseAttribute.boolean('hideendscreen'),
+      mediaEmbed: parseAttribute.string('mediaEmbed'),
+      extendedrelated: parseAttribute.boolean('extendedrelated'),
+      skin: parseAttribute.json('skin'),
+      showlogo: parseAttribute.boolean('showlogo'),
+      watermark: parseAttribute.boolean('watermark'),
+      qoeEventsConfig: parseAttribute.json('qoeEventsConfig'),
+      advVastDuration: parseAttribute.number('advVastDuration'),
+      vastTag: parseAttribute.string('vastTag'),
+      embedTrackings: parseAttribute.json('embedTrackings'),
+      id: parseAttribute.string('id'),
+
+      autoplay: parseAttribute.boolean(VideoAttributes.AUTOPLAY) || false,
+      ampnoaudio: parseAttribute.boolean(VideoAttributes.NO_AUDIO),
+      dock: parseAttribute.boolean(VideoAttributes.DOCK),
+      rotateToFullscreen: parseAttribute.boolean(
+          VideoAttributes.ROTATE_TO_FULLSCREEN,
+      ),
+    };
+  }
+
+  /**
+   * Method that sends postMessage to iframe that contains the player.
+   * Message is prepended with proper header.
+   * If the frame is not ready all messages will be saved in queue and
+   * sent whe runQueue method is called.
+   * @private
+   * @param {string} name name of the command
+   * @param {string} data optional data to send with the command
+   * @param {boolean} skipQueue if this parameter is present the message will
+   * send the command even when the frame is not read
+   */
+  sendCommand_(name, data, skipQueue = false) {
+    if (this.frameReady_ || skipQueue) {
+      this.contentWindow_.postMessage(data
+        ? `${this.header_}${name}@PAYLOAD@${data}`
```

Comment:
@aghassemi Could there be security concerns with sending over data via `postMessage` in a non-standard format?